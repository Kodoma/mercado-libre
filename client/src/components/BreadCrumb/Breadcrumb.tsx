import React from "react";
import styles from "./Breadcrumb.module.css";
import fetch from "cross-fetch";

const Breadcrumb: React.FunctionComponent<BreadCrumbProps> = ({search}) => {
    const [categories, setCategories] = React.useState(CategoriesInitState);

    React.useEffect(() => {
        async function fetchItems(search: string) {
            if(!search) return;
            const request = await fetch(
                "http://0.0.0.0:8080/api/items?search="+search
            );
            const response = await request.json();
            setCategories(response.categories);
        }

        fetchItems(search as string);
    },[search]);

    const breadcrumbs = categories.join(' > ');

    return (
        <div className={styles.breadcrumbs}>
            { breadcrumbs }
        </div>
    )
}

export default Breadcrumb

interface BreadCrumbProps {
    search: string;
}

const CategoriesInitState: string[] = [];