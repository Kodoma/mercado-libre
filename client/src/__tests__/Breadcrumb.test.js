import React from 'react';
import 'jsdom-global/register';
import {mount, shallow} from 'enzyme';
import Breadcrumb from '../components/BreadCrumb/Breadcrumb';

describe('__tests__ Breadcrumb', () => {
    it('should render my Breadcrumb component', () => {
        const component = shallow(<Breadcrumb search={'iphone'} />);
        expect(component.getElements()).toMatchSnapshot();
    });

    it('should render empty Breadcrumb', function () {
        const wrap = mount(<Breadcrumb search={''} />)
        expect(wrap.find('.breadcrumbs').find('ol').text()).toBe('')
    })
    
});
