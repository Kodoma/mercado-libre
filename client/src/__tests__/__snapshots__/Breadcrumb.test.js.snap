// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests Breadcrumb should render my Breadcrumb component 1`] = `
Array [
  <div
    className="breadcrumbs"
  >
    
  </div>,
]
`;
